core/src/main/scala/org/genivi/sota/core/transfer/VehicleUpdates.scala

```diff
@@ -125,36 +126,67 @@ object VehicleUpdates {
       .update(UpdateStatus.Canceled)
   }
 
+  /**
+    * Find the [[UpdateRequest]]-s for:
+    * <ul>
+    *   <li>the given VIN</li>
+    *   <li>and for which [[UpdateSpec]]-s exist with Pending or InFlight [[UpdateStatus]]</li>
+    * </ul>
+    */
   def findPendingPackageIdsFor(vin: Vehicle.Vin)
                               (implicit db: Database, ec: ExecutionContext) : DBIO[Seq[UpdateRequest]] = {
     updateSpecs
       .filter(_.vin === vin)
       .filter(_.status.inSet(List(UpdateStatus.InFlight, UpdateStatus.Pending)))
       .join(updateRequests).on(_.requestId === _.id)
-      .sortBy(r => (r._1.installPos.asc, r._2.creationTime.asc))
-      .map { case (sp, ur) => (sp.installPos, ur) }
+      .sortBy { case (sp, _) => (sp.installPos.asc, sp.creationTime.asc) }
+      .map    { case (_, ur) => ur }
       .result
-      .map { _.map { case (idx, ur) => ur.copy(installPos = idx) } }
+      .map { _.zipWithIndex.map { case (ur, idx) => ur.copy(installPos = idx) } }
   }
 
   /**
-    * TODO FIXME the returned [[UpdateSpec]] doesn't include real dependencies: just an empty set.
+    * Find the [[UpdateSpec]]-s (including dependencies) whatever their [[UpdateStatus]]
+    * for the given ([[UpdateRequest]], VIN) combination.
+    * <br>
+    * Note: for pending ones see [[findPendingPackageIdsFor()]]
     */
   def findUpdateSpecFor(vin: Vehicle.Vin, updateRequestId: UUID)
                        (implicit ec: ExecutionContext, db: Database): DBIO[UpdateSpec] = {
-    updateSpecs
+
+    val opt: DBIO[Option[((Namespace, UUID, Vin, UpdateStatus, Int, DateTime), UpdateRequest)]] =
+      updateSpecs
       .filter(_.vin === vin)
       .filter(_.requestId === updateRequestId)
       .join(updateRequests).on(_.requestId === _.id)
       .result
       .headOption
-      .flatMap {
+
+    opt.flatMap {
         case Some((
           (ns, uuid, updateVin, status, installPos, creationTime),
           updateRequest
           )) =>
-          val spec = UpdateSpec(updateRequest, updateVin, status, Set.empty[Package], installPos, creationTime)
-          DBIO.successful(spec)
+
+          val reqPaks0: DBIO[Seq[(Namespace, UUID, Vin, Name, Version)]] =
+            requiredPackages
+            .filter(_.namespace === ns)
+            .filter(_.requestId === updateRequestId)
+            .filter(_.vin === vin)
+            .result
+
+          val reqPaks: DBIO[Seq[PackageId]] = reqPaks0 map { rows =>
+            rows map { case (_, _, _, pakName, pakVersion) =>
+              PackageId(pakName, pakVersion) }
+          }
+
+          val paksDB: DBIO[Seq[Package]] = for (
+            pakIds <- reqPaks;
+            paks   <- Packages.byIds(ns, pakIds.toSet)
+          ) yield paks
+
+          paksDB map { rows => UpdateSpec(updateRequest, updateVin, status, rows.toSet, installPos, creationTime) }
+
```

Comment:
Docu before and after, for this method:
```diff
-    * TODO FIXME the returned [[UpdateSpec]] doesn't include real dependencies: just an empty set.
+    * Find the [[UpdateSpec]]-s (including dependencies) whatever their [[UpdateStatus]]
+    * for the given ([[UpdateRequest]], VIN) combination.
```
This PR assume we'd like to know the real (as stored in the DB) dependencies.

It's long winded because of the way information is split across the `UpdateRequest`, `UpdateSpec`, and `Requirements` tables.